### PR TITLE
blockchain/v2: add validate basic to blocks

### PR DESCRIPTION
**Pull request by cmwaters**
## Description

Closes: #5396

### CODE REVIEW

File: types/validator_set.go [hunk 1]
```diff
@@ -718,6 +718,9 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID,
 // signatures.
 func (vals *ValidatorSet) VerifyCommitLight(chainID string, blockID BlockID,
 	height int64, commit *Commit) error {
+	if commit == nil {
```
Comment (melekes) [inline types/validator_set.go:721]: do we want to add this to `VerifyCommit` as well?

Comment (cmwaters) [reply types/validator_set.go:721]: Yeah that probably makes sense. (potentially also `VerifyCommitLightTrusting`)

**Review by erikgrinaker — APPROVED** [review]

File: blockchain/v2/scheduler.go [hunk 2]
```diff
@@ -522,6 +522,12 @@ func (sc *scheduler) handleBlockResponse(event bcBlockResponse) (Event, error) {
 		return scPeerError{peerID: event.peerID, reason: err}, nil
 	}
 
+	err = event.block.ValidateBasic()
```
Comment (marbar3778) [inline blockchain/v2/scheduler.go:525]: Do we want this check to happen in v0 and v1 as well?

**Comment by cmwaters** [comment]
I missed that we do a `ValidateBasic()` when we convert from proto therefore I'm going to close this PR. I could keep the nil checks in `VerifyCommitLight()` but as far as I know we always run `ValidateBasic()` to check that the commit is not nil before running `VerifyCommitLight()` so it's not necessary. Let me know if anyone disagrees.

**Pull request closed without merging**